wordpress-mobile/WordPress-Android
# Gutenberg : Html mode switching after image upload cancellation

Fixes # https://github.com/wordpress-mobile/gutenberg-mobile/issues/2048

## Findings
Based on my investigation it seemed the root cause of this issue involved the `localMediaId` being used to track the progress of each upload being removed as the wrong type from the `mUploadingMediaProgressMax`ConcurrentHashMap which caused no removal at all. Also, when the upload is canceled via the dialog, the `EditPostActivity` would still send another progress event after the cancellation which would put a progress value back into the `mUploadingMediaProgressMax`ConcurrentHashMap so even though the positive action of the cancel dialog was removing the progress value for the `localMedia` id from the map it would be put back in. Hence when you clicked on the `Html mode` action in the overflow menu, it would still think that's there's still an upload in progress even though it was canceled.

## Solution
The solution was for this issue was two-fold. First, the `localMediaIds` being used as as `Integers` in places that were expecting a `String` were converted to a `String` and a `Set<String>` was incorporated to track the media ids that were canceled so progress events for them could be ignored. 

Further comments annotate the code below.

## Testing

Before testing, you can lessen the network speed of your emulator by clicking on the menu option of the emulator, and choosing the lowest Network type and Signal strength to ensure the upload doesn't happen too quickly. 

1. Start a Gutenberg post and add an image block.
2. Add media by selecting "Choose from device". Before the upload finishes tap on the image to trigger the "Stop uploading" dialog and select "yes" to cancel the upload.
3. Tap on the 3-dot menu and select the "HTML mode" option
4. The editor changes mode and a toast message come up to announce that the switch occurred.

<kbd><img src="https://user-images.githubusercontent.com/1509205/96039900-054b3a00-0e2f-11eb-817e-b0ca0f3f4cef.gif" width="320"></kbd>

-----------------

1. Start a Gutenberg post and add an image block.
2. Add media by selecting "Choose from device". Before the upload finishes tap on the 3-dot menu and select the "HTML mode" option.
3. The editor should show a toast message that the switch can't occur during upload. 

<kbd><img src="https://user-images.githubusercontent.com/1509205/96040086-54916a80-0e2f-11eb-8999-bbb19b1b59ce.gif" width="320"></kbd>

------------------

1. Start a Gutenberg post and add an image block.
2. Add media by selecting "Choose from device". 
3.After the upload finishes tap on  the 3-dot menu and select the "HTML mode" option
4. The editor changes mode and a toast message come up to announce that the switch occurred.

<kbd><img src="https://user-images.githubusercontent.com/1509205/96040274-a2a66e00-0e2f-11eb-8ebd-eea19cd07a49.gif" width="320"></kbd>

## Reviewing 
Only 1 reviewer is needed but anyone can review. 

## Submitter Checklist
